### PR TITLE
Expose a way to create a utf8z from a ReadOnlySpan<byte>

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@ericsink I'm also thinking about exposing `util.to_utf8_with_z` as a static method off of utf8z (with a more appropriate .net name).  Basically, it would just ensure that clients of the library agree with the library on exactly how a string should be encoded to be valid for you. 

Would that be ok with you?

**Comment by CyrusNajmabadi** [comment]
@ericsink ?

**Comment by ericsink** [comment]
Yeah, stand by, I'm trying to catch up with this.  :-)

**Comment by ericsink** [comment]
LGTM.  In fact, you've gone rather above and beyond the call of duty here with the API comments, thanks.

The `util.to_utf8_with_z` change you mentioned would be fine too.  Should I merge this PR now or wait for that piece?

**Comment by CyrusNajmabadi** [comment]
I'll add that piece.

**Comment by CyrusNajmabadi** [comment]
@ericsink ready.

File: src/SQLitePCLRaw.core/utf8z.cs [hunk 1]
```diff
@@ -138,7 +173,19 @@ public string utf8_to_string()
...
+        /// <summary>
+        /// Gets the <see cref="Encoding.UTF8"/> encoded bytes for the provided <paramref name="value"/>.  The array
+        /// will include a trailing <c>\0</c> character.  The length of the array will <see cref="Encoding.UTF8"/>'s
+        /// <see cref="Encoding.GetByteCount(string)"/><c>+1</c> (for the trailing <c>\0</c> byte).  These bytes are
+        /// suitable to use with <see cref="FromSpan"/> using the extension <see
+        /// cref="MemoryExtensions.AsSpan{T}(T[])"/> or <see cref="FromPtr(byte*)"/> or <see cref="FromPtrLen(byte*,
+        /// int)"/>.  Note that for <see cref="FromPtrLen(byte*, int)"/> the length provided should not include the
+        /// trailing <c>\0</c> terminator.
+        /// </summary>
+        public static byte[] GetNullTerminatedUTF8Bytes(string value)
```
Comment (CyrusNajmabadi) [inline src/SQLitePCLRaw.core/utf8z.cs:186]: alternate name: GetZeroTerminatedUTF8Bytes.

Comment (ericsink) [reply src/SQLitePCLRaw.core/utf8z.cs:186]: I almost suggested that.  

I do somewhat prefer "zero-terminated" to "null-terminated", even though the latter is far more common usage.  I like to reserve the term "null" for pointers, references, the absence of a value in SQL, etc.  The zero-byte at the end of a C string was originally called NUL in ASCII, so at some point people spoke of a NUL-terminated string, and then NUL and null got conflated.  Usually when I go down this path I tell myself I'm being too pedantic.

Comment (CyrusNajmabadi) [reply src/SQLitePCLRaw.core/utf8z.cs:186]: or: GetUTF8BytesWithTrailingZero

Comment (ericsink) [reply src/SQLitePCLRaw.core/utf8z.cs:186]: Not bad, but it reminds me of `System.Numerics.BitOperations.TrailingZeroCount`.

I don't see a slam-dunk winner here.  Either "ZeroTerminated" or "NullTerminated" would be fine, but given my slight preference for the former, if you don't mind, perhaps I'll go ahead and change it to `GetZeroTerminatedUTF8Bytes`.

Comment (CyrusNajmabadi) [reply src/SQLitePCLRaw.core/utf8z.cs:186]: Whatever works best for you :)

**Pull request merged**